crates/orderbook/src/database/orders.rs

```diff
@@ -179,18 +176,24 @@ impl OrderStoring for Postgres {
 
         let order = order.clone();
         let mut connection = self.pool.acquire().await?;
-        connection
-            .transaction(move |transaction| {
-                async move {
-                    insert_order(&order, transaction).await?;
-                    if let Some(quote) = quote {
-                        insert_quote(&order.metadata.uid, &quote, transaction).await?;
-                    }
-                    Ok(())
-                }
-                .boxed()
-            })
-            .await
+        let mut ex = connection.begin().await.unwrap();
+
+        insert_order(&order, &mut ex).await?;
+        if let Some(quote) = quote {
+            insert_quote(&order.metadata.uid, &quote, &mut ex).await?;
+        }
+        if let Some(full_app_data) = order.metadata.full_app_data {
+            let contract_app_data = &ByteArray(order.data.app_data.0);
+            let full_app_data = full_app_data.as_bytes();
+            let existing =
+                database::app_data::insert(&mut ex, contract_app_data, full_app_data).await?;
+            if full_app_data != existing {
+                return Err(InsertionError::AppDataMismatch(existing));
+            }
+        }
+
+        ex.commit().await.unwrap();
```

Comment:
The function already returns a `Result`. Maybe a failing `commit` should return an error as well instead of being unwrapped.

Reply:
Good catch. These shouldn't unwrap at all. I don't know what I was thinking when I added unwrap here.